proposals/0000-add-support-for-css-color-function.md

```diff
@@ -0,0 +1,109 @@
+
+3. Update [RCTTextPrimitivesConversions](https://github.com/facebook/react-native/blob/ac1cdaa71620d5bb4860237cafb108f6aeae9aef/packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextPrimitivesConversions.h#L116) to handle setting new color values for color space. If color space is not included preserve existing behavior.
+
+```objc
+[UIColor colorWithDisplayP3Red:components.red green:components.green blue:components.blue alpha:components.alpha]
+```
+
+### Android Changes
+
+1. Enable the wide color gamut in [ReactActivity](https://github.com/facebook/react-native/blob/7625a502960e6b107e77542ff0d6f40fbf957322/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactActivity.java#L22) if available.
+
+```java
+Window window = getWindow();
+boolean isScreenWideColorGamut = window.getDecorView().isScreenWideColorGamut()
+if (isScreenWideColorGamut) {
+  window.setColorMode(PixelFormat.COLOR_MODE_WIDE_COLOR_GAMUT)
+}
+```
+
+2. Update [ColorUtil](https://github.com/facebook/react-native/blob/a6964b36294c3bfea09c0cdd65c5d0e3949f2dae/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ColorUtil.java#L17) and [ColorPropConverter](https://github.com/facebook/react-native/blob/781b637db4268ad7f5f3910d99ebb5203467840b/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/ColorPropConverter.java#L27) to return new color values for color space as longs.
+
+```java
+@ColorLong long p3 = pack(1.0f, 1.0f, 0.0f, 1.0f, ColorSpace.Named.DISPLAY_P3);
+Color opaqueYellow = Color.valueOf(p3);
+```
+
+### Docs
+
+Update the [Color Reference](https://github.com/facebook/react-native-website/blob/main/docs/colors.md) to document color function usage in React Native.
+
+## Drawbacks
+
+There should be no breaking changes for users but the color implementation will be a bit more complicated than before.
+
+## Alternatives
+
+We could do nothing and require users to patch package for Display P3 color support. But this is a disappointing developer experience and not a satisfying solution.
+
+## Adoption strategy
+
+This is intended to be a non-breaking change with zero impact to existing users. Adoption is entirely opt-in by using the color function in styles to start using display-p3 colors.
+
+## How we teach this
+
+We should document the inclusion of the color function in the [official color reference](https://reactnative.dev/docs/colors). Developers can then start using it if they choose.
+
+## Unresolved questions
+
+- While Android does support wide color gamut, do [Android View](<https://developer.android.com/reference/android/view/View#setBackgroundColor(int)>)'s actually support it? These methods take `int`s while the Android color reference states wide gamut colors are `long`s.
```

Comment:
Answer to the first: no, of course. We would have to update the signatures that receives an int.
This would be a breaking change, so we need to look for overloading those functions :( 

This bit needs to be moved above in the implementation details.

Reply:
@cipolleschi I'm not seeing an alternative in the Android documentation to set a View's background color to a long though. Looking at https://developer.android.com/training/wide-color-gamut it only mentions images and using OpenGL or Vulkan. It doesn't seem like Android's View supports DisplayP3.